### PR TITLE
improve link integrity popup

### PR DESCRIPTION
**Pull request by giuliaghisini**
fixed and improved link integrity popup when deleting item/items from folder contents. 
Moved all code in a separate component 'ContentsDeleteModal'.

Added loader when loading references. If the call to @linkintegrity took a long time to load, it wasn't clear to the user that it was checking for broken references:
 
loading

If only one item is selected to delete, it will display all the possibile broken references in a table, instead in a list. In the last column could be more than one reference: 
1-selected-item

If more than one item is selected to delete, it will not display the possibile broken references, but only the number, because it  will be a very long list (It worked like that before too):
more-than-one-selected

### CODE REVIEW

File: packages/volto/locales/volto.pot [hunk 1]
```diff
@@ -2355,6 +2356,11 @@ msgstr ""
 msgid "Navigate back"
 msgstr ""
 
+#. Default: "Navigate to this item"
+#: components/manage/Contents/ContentsDeleteModal
+msgid "Navigate to this items"
```
Comment (stevepiercy) [inline packages/volto/locales/volto.pot:2361]: This should be either "Navigate to this item" or the "Navigate to these items", and align with the Default. I'm not sure whether it should be plural or singular. Please advise.

File: packages/volto/locales/volto.pot [hunk 2]
```diff
@@ -4690,6 +4706,21 @@ msgstr ""
...
+#: components/manage/Contents/ContentsDeleteModal
+msgid "link-integrity: Delete"
+msgstr ""
+
+#. Default: "Delete item and break links"
+#: components/manage/Contents/ContentsDeleteModal
+msgid "link-integrity: Delete item and break links"
+msgstr ""
+
+#. Default: "Checking references..."
```
Comment (stevepiercy) [inline packages/volto/locales/volto.pot:4719]: Let's not introduce punctuation that might not be translatable.

```suggestion
#. Default: "Checking references"
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
I only reviewed the message strings and defaults. This still requires a technical review. But it looks pretty good.

**Review by stevepiercy — APPROVED** [review]
Messages only approved. Still requires technical review.

**Comment by sneridagh** [comment]
@giuliaghisini we merged #6509 so now importing from barrel files in core is not allowed anymore, so expect some ESlint failures. Could you please update the PR accordingly? Thanks!

File: packages/volto/.env.local
Comment (pnicolli) [inline packages/volto/.env.local:1]: Remove this file (maybe we should gitignore *.local files? @sneridagh )

**Review by davisagli — APPROVED** [review]
@giuliaghisini Nice improvements, thank you.

**Pull request merged**